### ISSUE TITLE
Unreasonably High Loss with Sample Packing on Qwen3Moe

### ISSUE THREAD

**Issue by nyxkrage** [axolotl-ai-cloud/axolotl#2626]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

I would expect the difference in loss between having sample_packing on and off to be minimal and not 3x as high with 4096 sequence_length

### Current behaviour

Currently with (eval_)sample_packing off, the loss & eval loss sits at around 1-1.5, whereas without it train loss sits at around 3-4 and eval loss at around 4.5.

I am aware that the Qwen3 15B MoE is private, however trying to reproduce on the 30B results Out Of Memory errors, or if trying to use Zero2, the following error `Error invalid configuration argument at line 233 in file /src/csrc/ops.cu`. If needed I can share the weights of the 15B model.

### Steps to reproduce

1. Run the given config as is, with `sample_packing` and `eval_sample_packing` set to true and 
1. Observe loss 
1. Disable sample packing and rerun it
1. Observe loss is now much lower than before

### Config yaml

```yaml
base_model: Qwen3-15B-A2B-Base
model_type: AutoModelForCausalLM
tokenizer_type: AutoTokenizer

load_in_8bit: false
load_in_4bit: false
strict: false

datasets:
  - path: NewEden/Claude-Instruct-2.7K
    type: chat_template
    roles_to_train: ["gpt"]
    field_messages: conversations
    message_field_role: from
    message_field_content: value
    train_on_eos: turn
shuffle_merged_datasets: true
dataset_prepared_path: dataset_preparedss
val_set_size: 0.1
output_dir: qwen-dbg

sequence_len: 4096
sample_packing: true
eval_sample_packing: true
pad_to_sequence_len: true

evals_per_epoch: 2
eval_table_size:
eval_max_new_tokens: 128

gradient_accumulation_steps: 1
micro_batch_size: 1
num_epochs: 1
optimizer: adamw_bnb_8bit
lr_scheduler: cosine
learning_rate: 3e-6

train_on_inputs: false
group_by_length: false
bf16: auto
fp16:
tf32: false

gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true
s2_attention:

warmup_steps: 40
saves_per_epoch: 2
debug:
deepspeed:
weight_decay: 0.0
fsdp:
fsdp_config:
special_tokens:
```

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10.12

### axolotl branch-commit

ed922796b71bc38c2504b676b445da14fa932ada

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by nyxkrage** [comment]
Can't be reproduced by NanoBit, so I'll assume something went wrong when I updated to ed922796b71bc38c2504b676b445da14fa932ada, and re-open if it turns out to still be the case.
